scripts/checkArticles.js

```diff
@@ -0,0 +1,194 @@
+  }
+}
+
+async function fetchAllArticles() {
+  const query = {
+    page: 1,
+    'page-size': 100,
+  };
+  const firstResult = await fetchArticles(query, token);
+
+  const numberOfPages = Math.ceil(
+    firstResult.totalCount / firstResult.pageSize,
+  );
+  const requests = [];
+  const estimatedTime = (numberOfPages * query['page-size'] * 0.1) / 60;
+
+  console.log(
+    `Fetching ${numberOfPages} pages with a page size of ${
+      query['page-size']
+    }. Estimated time is ${estimatedTime} minutes`,
+  );
+
+  for (let i = 1; i < numberOfPages + 1; i += 1) {
+    requests.push(fetchArticles({ ...query, page: i }, token));
+    await sleep(500); // eslint-disable-line
+
+    console.log(
+      `${chalk.green(
+        `Fetching page ${i} with page size ${query['page-size']}`,
+      )}`,
+    ); //eslint-disable-line
+  }
+
+  const articleIds = [];
+  const results = await Promise.all(requests);
+  results.forEach(result => {
+    const articles = result ? result.results.map(article => article.id) : [];
+    articleIds.push(...articles);
+  });
+  return articleIds;
+}
+
+async function testArticle(id, article) {
+  try {
+    if (article) {
+      const converted = learningResourceContentToEditorValue(
+        article.content.content,
+        fragment,
+      );
+      learningResourceContentToHTML(converted);
+      console.log(
+        `${chalk.green(
+          `Article with id ${id} was sucessfully converted to slate and back. The article id is: ${
+            article.id
+          }`,
+        )}`,
+      );
+    }
+  } catch (err) {
+    errors.push({ error: err, id });
+    console.log(
+      `${chalk.red(`Article with id ${id} failed to convert.`)}`,
+      err,
+    ); //eslint-disable-line
+  }
+}
+
+async function runCheck() {
+  await fetchSystemAccessToken();
+  const readFromFile = process.argv[2] !== '-write';
+  if (!readFromFile) {
+    const articles = [];
+    const articleIds = await fetchAllArticles();
+    await asyncForEach(articleIds, async id => {
+      const article = await fetchArticle(id, token);
+      articles.push(article);
+      await testArticle(id, article);
+    });
+    fs.writeFileSync(
+      './scripts/articles.json',
+      JSON.stringify(articles),
+      'utf8',
+    );
+  } else {
+    const path = './scripts/articles.json';
+    if (!fs.existsSync(path)) {
+      console.log(
+        `${chalk.red(
+          `Run yarn check-articles-and-write instead. File with path ${path} does not exist.`,
+        )}`,
+      ); //eslint-disable-line
+      return;
+    }
+    const contents = fs.readFileSync(path);
+    const articles = await JSON.parse(contents);
+
+    articles.forEach(article => {
+      if (article) testArticle(article.id, article);
+    });
+  }
+  console.log(
+    `Total errors: ${errors.length}. Articles that is failing is: ${
+      errors ? errors.map(error => (error ? error.id : '')) : '[]'
+    }`,
+  ); //eslint-disable-line
```

Comment:
Synes det er mange `eslint-disable-line` nedover. Hva med å legge det på toppen og disable for hele dokumentet? Og gjerne hva man disabler.

Reply:
Fikset